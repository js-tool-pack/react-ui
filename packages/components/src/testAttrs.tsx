import { fireEvent, render } from '@testing-library/react';
import type { PropsBase } from '@pkg/shared';
import React from 'react';

export function testAttrs<
  T extends { attrs?: Partial<React.HTMLAttributes<HTMLElement>> },
>(component: React.FC<T>, displayName?: string) {
  const name = displayName || component.displayName || component.name;
  test(name ? `${name} attrs` : `attrs`, () => {
    const onClick = jest.fn();

    const FC = component as React.FC<PropsBase>;
    const { container } = render(
      <FC
        attrs={{
          style: { width: '20px' },
          className: 'foo',
          role: 'note',
          onClick,
        }}
      />,
    );

    expect(container.firstChild).toHaveClass('foo');
    expect(container.firstChild).toHaveAttribute('role', 'note');
    expect(container.firstChild).toHaveStyle({ width: '20px' });

    expect(onClick).not.toBeCalled();
    fireEvent.click(container.firstChild!);
    expect(onClick).toBeCalled();
  });
}
