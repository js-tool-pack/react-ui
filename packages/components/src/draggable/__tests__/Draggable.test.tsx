import { render } from '@testing-library/react';
import { testAttrs } from '~/testAttrs';
import { useState } from 'react';
import { Draggable } from '..';

describe('Draggable', () => {
  testAttrs(Draggable);
  test('base', () => {
    const App = () => {
      const [state, setState] = useState<{ name: string; id: number }[]>([
        { name: 'John', id: 1 },
        { name: 'Joao', id: 2 },
        { name: 'Jean', id: 3 },
      ]);
      return (
        <Draggable onChange={setState} list={state}>
          {state.map((item, index) => (
            <div className="draggable-item" key={item.id}>
              <span>{index + 1}.</span> <span>{item.name}</span>{' '}
              <span>{item.id}</span>
            </div>
          ))}
        </Draggable>
      );
    };
    const r = render(<App />);
    expect(r.container.firstChild).toMatchSnapshot();
  });
});
