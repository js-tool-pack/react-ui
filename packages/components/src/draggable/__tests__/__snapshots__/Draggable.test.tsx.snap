// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Draggable base 1`] = `
<div
  class="t-draggable"
>
  <div
    class="draggable-item"
  >
    <span>
      1
      .
    </span>
     
    <span>
      John
    </span>
     
    <span>
      1
    </span>
  </div>
  <div
    class="draggable-item"
  >
    <span>
      2
      .
    </span>
     
    <span>
      Joao
    </span>
     
    <span>
      2
    </span>
  </div>
  <div
    class="draggable-item"
  >
    <span>
      3
      .
    </span>
     
    <span>
      Jean
    </span>
     
    <span>
      3
    </span>
  </div>
</div>
`;
